Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py

```diff
@@ -0,0 +1,151 @@
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import logging
+import re
+import sys
+from xml.etree.ElementInclude import include
+
+from .command import Command
+from .branch import Branch
+from.pull_request import PullRequest
+
+from webkitbugspy import Tracker
+from webkitcorepy import arguments, run, Terminal
+from webkitscmpy import local, log, remote
+
+
+class Revert(Command):
+    name = 'revert'
+    help = 'Revert provided list of commits and create a pull-request with this revert commit'
+
+    @classmethod
+    def parser(cls, parser, loggers=None):
+        PullRequest.parser(parser, loggers=loggers)
+        # Only allow revert one commit one time, because git automatically generate mesaage will only contain one commit information
+        parser.add_argument(
+            'commit',
+            help='git hash, svn revision or identifer you want to revert'
+        )
+
+    @classmethod
+    def revert_commit(cls, args, repository, **kwargs):
+        # Check if there are any outstanding changes:
+        if repository.modified():
+            sys.stderr.write('Please commit your changes or stash them before you revert commit: {}'.format(args.commit))
+            return 1
+        # Make sure we have the commit that user want to revert
+        try:
+            commit = repository.find(args.commit, include_log=False)
+        except (local.Scm.Exception, ValueError) as exception:
+            # ValueErrors and Scm exceptions usually contain enough information to be displayed
+            # to the user as an error
+            sys.stderr.write(str(exception) + '\n')
+            return 1
+
+        result = run([repository.executable(), 'revert', '--no-commit'] + [commit.hash], cwd=repository.root_path, capture_output=True)
+        if result.returncode:
+            # git revert wiil output nothing if this commit is already reverted
+            if not result.stdout.strip():
+                sys.stderr.write('The commits you spiecfied seems already be reverted.')
+                return 2
+            # print out
+            sys.stderr.write(result.stdout.decode('utf-8'))
+            sys.stderr.write(result.stderr.decode('utf-8'))
+            log.info('If you have merge conflicts, after resolving them, please use git-webkit pfr to publish your pull request')
```

Comment:
I think this should be a `sys.stderr.write`, `log.info` will be silenced by `-q`, which is probably not desirable